chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -610,33 +598,37 @@ trait TransformerMacros extends TransformerConfigSupport with MappingMacros with
                   Right(cq"_: $instTpe => $instanceTree")
                 }
                 .getOrElse {
+                  // in simple scenarios left-hand side of the case expression is flat (`case _: $type` or `case $value`)
+                  val flatPatternMatch: Tree =
```

Comment:
This is the most important change I'd like to get your feedback for.

Instead of a single match-case clause covering all possible pairs of `From, To`, I split it into 2 expressions.

The first one calculates a pattern match expression (left-hand side of the case clause), and the other calculates entire tree by (mostly) inspecting the matching target symbol.

`if` guards are moved into extractors to allow for shorter pattern matching.